### PR TITLE
Ta remove questions options

### PR DESCRIPTION
**Pull request by NEUDitao**
it works i swear @liustanley

### CODE REVIEW

File: packages/app/components/Queue/TAQueueList.tsx [hunk 1]
```diff
@@ -246,7 +246,9 @@ interface QueueProps {
 function QueueQuestions({ questions, isHelping, onOpenCard }: QueueProps) {
   return (
     <div data-cy="queueQuestions">
-      {questions.length === 0 ? (
+      {questions.filter(
+        (question) => question.status !== OpenQuestionStatus.TADeleted
+      ).length === 0 ? (
```
Comment (liustanley) [inline packages/app/components/Queue/TAQueueList.tsx:251]: Could store this filtered questions array into a constant and just use this both places

File: packages/app/components/Queue/StudentQueueList.tsx [hunk 2]
```diff
@@ -285,7 +301,9 @@ interface QueueProps {
 function QueueQuestions({ questions, studentQuestion }: QueueProps) {
   return (
     <div data-cy="queueQuestions">
-      {questions?.length === 0 ? (
+      {questions?.filter(
+        (question) => question.status !== OpenQuestionStatus.TADeleted
+      ).length === 0 ? (
```
Comment (liustanley) [inline packages/app/components/Queue/StudentQueueList.tsx:306]: Same thing here, store filter result into a constant

File: packages/app/components/Queue/StudentPopupCard.tsx [hunk 3]
```diff
@@ -117,7 +112,8 @@ const StudentPopupCard = ({
               cancelText="No"
               onConfirm={() => {
                 onClose();
-                updateQuestion(question, ClosedQuestionStatus.Deleted);
+                updateQuestion(question, OpenQuestionStatus.TADeleted);
+                API.questions.notify(question.id);
```
Comment (liustanley) [inline packages/app/components/Queue/StudentPopupCard.tsx:116]: This uses the can't find message right now for the notification, should be using the `REMOVED` message instead

File: packages/app/components/Queue/StudentRemovedFromQueueModal.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      visible={props.question?.status === OpenQuestionStatus.TADeleted}
+      footer={[
+        <Button key="leave" danger onClick={props.leaveQueue}>
+          Leave Queue
+        </Button>,
+        <Button type="primary" key="rejoin" onClick={props.joinQueue}>
+          Rejoin Queue
+        </Button>,
+      ]}
+    >
```
Comment (liustanley) [inline packages/app/components/Queue/StudentRemovedFromQueueModal.tsx:26]: Add `closable={false}` here to remove the modal X

File: packages/app/components/Queue/StudentRemovedFromQueueModal.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      You&quot;ve been removed from the queue by the following TA:{" "}
+      {props.question?.taHelped?.name}. If you have any questions, please reach
+      out to the TA. If you&quot;d like to join back into the queue with your
```
Comment (liustanley) [inline packages/app/components/Queue/StudentRemovedFromQueueModal.tsx:29]: The apostrophes are showing up as double quotes, pls fix

also change 'cancel' to 'leave queue'

**Review by dajinchu — COMMENTED** [review]
Seems to work locally. Not sure what the issue with studentQuestion was still?

File: packages/app/components/Queue/StudentRemovedFromQueueModal.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        <Button key="leave" danger onClick={props.leaveQueue}>
+          Leave Queue
+        </Button>,
+        <Button type="primary" key="rejoin" onClick={props.joinQueue}>
+          Rejoin Queue
+        </Button>,
+      ]}
+    >
+      You&quot;ve been removed from the queue by the following TA:{" "}
+      {props.question?.taHelped?.name}. If you have any questions, please reach
```
[the remark this answers is not in the document]
Comment (NEUDitao) [reply packages/app/components/Queue/StudentRemovedFromQueueModal.tsx:28]: Screen Shot 2020-09-03 at 11 24 47 AM

yhow could you DO THIS TO ME (sure)

[the remark this answers is not in the document]
Comment (liustanley) [reply packages/app/components/Queue/StudentRemovedFromQueueModal.tsx:28]: xd you were the one to make the button "Leave Queue" though (which i like more)

File: packages/server/src/question/question.controller.ts [hunk 7]
```diff
@@ -229,11 +229,18 @@ export class QuestionController {
       relations: ['queue'],
     });
 
-    // TODO: somehow store and check that the notifying TA is the one helping? new UnauthorizedException('Only TA can send alerts');
+    console.log(question);
```
Comment (liustanley) [inline packages/server/src/question/question.controller.ts:232]: remove dis

File: packages/app/components/Queue/TAQueueList.tsx (hunk 1 above)
Comment (NEUDitao) [reply packages/app/components/Queue/TAQueueList.tsx:251]: gonna handle this in my ticket w/ Olivia

**Review by liustanley — APPROVED** [review]
i approve for now but pls fix filter stuff nvm just read ur comment u good

File: packages/server/src/question/question.controller.ts (hunk 7 above)
Comment (NEUDitao) [reply packages/server/src/question/question.controller.ts:232]: crap i missed this LMAO

Comment (NEUDitao) [reply packages/server/src/question/question.controller.ts:232]: i'll get it with olivia

**Pull request merged**